### PR TITLE
Add macos signing template

### PR DESCRIPTION
**Pull request by TravisEz13**

### CODE REVIEW

File: template-compliance/macOS-sign.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+- pwsh: |
+    [string] $CertificateId = "${{ parameters.certificateId }}"
+    Write-Verbose "CertificateId - $CertificateId" -Verbose
+
+    [string] $VariableName = "EsrpJson"
+
+    [string] $SigningServer = '$(SigningServer)'
+    Write-Verbose "SigningServer - $SigningServer" -Verbose
+
+    $esrp = @(@{
```
Comment (JamesWTruher) [inline template-compliance/macOS-sign.yml:20]: style question about this
this needs to be an array?
would `[hashtable[]]$esrp = ` be better (and more consistent with the above code?

**Review by JamesWTruher — APPROVED** [review]

File: template-compliance/macOS-sign.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ steps:
     [string] $SigningServer = '$(SigningServer)'
     Write-Verbose "SigningServer - $SigningServer" -Verbose
 
-    $esrp = @(@{
+    [hashtable[]] $esrp = @(@{
```
Comment (JamesWTruher) [inline template-compliance/macOS-sign.yml:20]: strictly speaking, the `@(` is no longer needed unless there are multiple hashtables, and then a comma between hashtables should suffice.

Comment (TravisEz13) [reply template-compliance/macOS-sign.yml:20]: It was never needed but testing this takes a LONG time.  I'll file an issue.

Comment (TravisEz13) [reply template-compliance/macOS-sign.yml:20]: Filed an Issue:
https://github.com/PowerShell/Compliance/pull/21#pullrequestreview-544502249

**Pull request merged**